### PR TITLE
feat:[CDS-52332]: error framework integration for TFC

### PR DESCRIPTION
**Pull request by alebuh**
## Describe your changes
Add error handling for terraform cloud
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/44770)

### CODE REVIEW

**Comment by alebuh** [comment]
trigger messagecheck

**Comment by alebuh** [comment]
trigger messagecheck

**Review by tmacari-harness — CHANGES_REQUESTED** [review]
> Reviewed 10 of 11 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44770)* status: 11 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/terraformcloud/TerraformCloudExceptionConstants.java` line 54 at r1](https://reviewable.io/reviews/harness/harness-core/44770#-NPvkONb8fm61bFeFI6c:-NPvkONb8fm61bFeFI6d:bqsye26) ([raw file](https://github.com/harness/harness-core/blob/1077ffb14bcb327fbe9d250f02ea6a173fd35cea/930-delegate-tasks/src/main/java/io/harness/delegate/task/terraformcloud/TerraformCloudExceptionConstants.java#L54)):*
> ```Java
> 
>   public static final class Message {
>     public static final String ERROR_GETTING_WORKSPACE = "Failed to to get a workspaces";
> ```

Failed to get a workspaces

**Review by tmacari-harness — DISMISSED** [review]
> Reviewed 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44770)* status: 11 file(s) to review. 1 of 2 approvals obtained.

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/terraformcloud/TerraformCloudTaskNG.java [hunk 1]
```diff
@@ -91,81 +104,88 @@ public DelegateResponseData run(TaskParameters parameters) throws IOException {
...
+                  taskParameters, commandUnitsProgress);
+          break;
+        default:
+          throw new InvalidRequestException("Terraform Cloud Task type not identified");
+      }
+      taskResponse.setUnitProgressData(UnitProgressDataMapper.toUnitProgressData(commandUnitsProgress));
+      taskResponse.setCommandExecutionStatus(CommandExecutionStatus.SUCCESS);
+      return taskResponse;
+    } catch (Exception e) {
+      TaskExceptionUtils.handleExceptionCommandUnits(
```
Comment (vlica-harness) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/terraformcloud/TerraformCloudTaskNG.java:164]: I believe we need to sanitise the exception here:
ExceptionMessageSanitizer.sanitizeException(e)
Also can we add an error log

**Review by alebuh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44770)* status: 11 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/terraformcloud/TerraformCloudTaskNG.java` line 164 at r2](https://reviewable.io/reviews/harness/harness-core/44770#-NPriGuoJlECIhjW0ckP-r2-164:-NPvyVqtAvmwbTc7Wy-_:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/dc8b773d01a718aee18f91198cddf89c892aa764/930-delegate-tasks/src/main/java/io/harness/delegate/task/terraformcloud/TerraformCloudTaskNG.java#L164)):*
<details><summary><i>Previously, vlica-harness (Vasile Lica) wrote…</i></summary><blockquote>

I believe we need to sanitise the exception here:
ExceptionMessageSanitizer.sanitizeException(e)
Also can we add an error log
</blockquote></details>

Done.

**Review by vlica-harness — APPROVED** [review]
> Reviewed 7 of 11 files at r1, 3 of 4 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44770)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by tmacari-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44770)* status: :shipit: complete! ✓

**Pull request merged**